### DISCUSSION TITLE
Rendering of the slice of a field

### DISCUSSION THREAD

**Discussion by remi-rc** [ProjectPhysX/FluidX3D#144] [Q&A] [answered]
Dear all,

I was wondering if there was an option to avoid the field-slice view to show very pixelated results, displaying the black background too much, which darkens the image ?

Currently, either in interactive or graphics mode, I have to slightly turn the view or unzoom a lot to obtain a "full" effect. Is it a rendering limitation, or could I simply set an option somewhere to get each cubic lattice to be fully colored ?

```
lbm.graphics.slice_mode = 2;
lbm.graphics.set_camera_centered(-90.0f, 0.0f, 0.0f, 4.5f);
lbm.graphics.write_frame(get_exe_path() + "export/camera/");
```

Is it be possible to slice through the TYPE_S as well ? If not, could this option be considered as a potential feature request ? It becomes important when considering internal flows, where the slice gets hidden by the geometry unless it is fully hidden.

Best,
Remi

**Comment by remi-rc** [comment]
Here is an image of what I obtain to make it easier to see the problem :)

![image](https://github.com/ProjectPhysX/FluidX3D/assets/137790854/3cacd16b-48da-48dd-83f3-32a2d5962243)

**Comment by ProjectPhysX** [comment]
Hi @remi-rc,

the `VIS_FIELD` mode draws a line of length 1 for each cell, in the direction of local velocity and with coloring by velocity magnitude. This is the same in slice mode, but then the lines become more obvious with more black space in between.
I've tought about adding a mode to draw filled suqares in slice mode instead of lines. I'll add this to the list of feature requests and leave this discussion open till then!

Kind regards,
Moritz

**Comment by remi-rc** [comment]
Hi @ProjectPhysX ,

thanks for adding this to the list :) 

Best,
Remi

**Comment by ProjectPhysX** [comment, marked as answer]
Hi @remi-rc,

please update, I've added filled slice rendering in [v2.14](https://github.com/ProjectPhysX/FluidX3D/releases/tag/v2.14)! Have fun!

Kind regards,
Moritz
